.github/ISSUE_TEMPLATE/bug_report.md

```diff
@@ -0,0 +1,44 @@
+<!--
+
+Have you read Formidable's Code of Conduct? By filing an Issue, you are expected to comply with it, including treating everyone with respect: https://github.com/FormidableLabs/spectacle/blob/master/CONTRIBUTING.md#contributor-covenant-code-of-conduct
+
+-->
+
+### Prerequisites
+
+**Feel free to delete this section if you have checked off all of the following.**
+
+- [ ] I've searched open [issues](https://www.github.com/FormidableLabs/spectacle/issues) to make sure I'm not opening a duplicate issue
+- [ ] This issue not specific to `spectacle-boilerplate` (those issues should be opened [here](https://github.com/FormidableLabs/spectacle-boilerplate/issues/new)).
+- [ ] I have read through the [docs](https://formidable.com/open-source/spectacle/docs/) before asking a question
+- [ ] I am using the latest version of Spectacle
+
+### Describe Your Environment
+
+What version of Spectacle are you using? (can be found by running `npm list spectacle`)
+
+What version of React are you using? (can be found by running `npm list react`)
+
+What browser are you using?
+
+What machine are you on?
+
+### Describe the Problem
+
+**Steps to reproduce:**
+
+1. [first step]
+2. [second step]
+3. [you get the idea]
```

Comment:
A lot of the time, we end up asking bug submitters for:

- A public minimal repository that we can `git clone`
- With install + error reproduction steps

This saves us an enormous amount of time from somewhat manual reproductions that can have different dependencies, etc.

I propose that we _default_ to asking for that or a WIP PR with a simple failing regression test. And I think that might be a good thing as a default for all our projects.

/cc @boygirl

Reply:
@ryan-roemer would you be okay with a codesandbox of the error or is that fall into the category of `somewhat manual reproduction`? If so, I'd be partial to a WIP PR - I think a separate repo could be a lot of overhead to submit an issue, especially for more jr devs

Reply:
@ryan-roemer Do you think the CodeSandbox link might also minimize some of this bug reproduction pain? We could require at least one of these three reproduction methods is included in every bug report, so we're not limiting anyone to use of one method but we ensure there is always a way to checkout the buggy code in the issue.

Also, I realize now I had the sandbox link at the bottom of this section; it would probably be better to reword this template so it's a more prominent requirement for a bug report.

Reply:
Let me back up. The _problem_ is that for a lot of repositories we get a report of a bug that looks like it may be a bug, but it isn't reproducible without a *lot* of work for the maintainers. Things that have cropped up in the past:

- Bespoke webpack config
- Dependencies variants
- Weird other libraries conflicting
- The bug not actually being a bug and something else being wrong.

For the maintainers, if something isn't immediately reproducible, that takes our valuable time trying to experiment and guess at how to set up the bug reproduction. That means trading off valuable time from everything else that we could otherwise focus on.

Contrast this with the bug opener. They're actually _hitting_ the bug. They have it! They are in the best, and most easy place to produce a working simulation (or the real repo if open source) for us of something that they can easily confirm is "the issue". On average, takes them way less time that us experimenting because they're already hitting the issue. So, the better balance if we have any resource constraints is to place a little more burden on the issue _openers_ than the maintainers so that the maintainers can field more bugs and work on fixing confirmed, reproducible bugs.

For jr devs, we most certainly can and should provide extra help with opening a failing WIP PR and/or reproduction. That's a good community investment too. But, I think we can roll in some language for that support while going for the overall goal of having "immediately reproducible" bugs that maintainers can easily verify. (It's just a goal -- won't always happen...)

So turning back to what we should ask for to meet our goal of "immediately reproducible" -- it's whatever does that! Things that should work:

- A failing regression test WIP PR
- A minimal repository with install + error reproduction instructions
- A code sandbox! (It has locked deps and is git cloneable, so it essentially **is** the above 😉 )
- ... other things!

Hope that helps!

Reply:
... and as a final thought -- not everyone has the time to do this / the capacity (for complicated stuff). And that's fine!

But, for stuff that _does_ have an immediately reproducible bug, that should influence us on what we fix first -- because it is something we can easily verify is wrong and much more readily fix. So it's not a **requirement** just a way for the community to help incentivize us to triage it higher priority / fix it sooner 😉